class ScopeTypeError(Exception):
    pass
class FunctionCalled(Exception):
    pass